allensdk/brain_observatory/behavior/project_apis/data_io/behavior_project_cloud_api.py

```diff
@@ -0,0 +1,127 @@
+    BehaviorProjectBase)
+from allensdk.brain_observatory.behavior.behavior_session import (
+    BehaviorSession)
+from allensdk.brain_observatory.behavior.behavior_ophys_session import (
+    BehaviorOphysSession)
+from allensdk.api.cloud_cache.cloud_cache import S3CloudCache
+
+
+class BehaviorProjectCloudApi(BehaviorProjectBase):
+    """API for downloading data released on S3
+     and returning tables.
+    """
+    def __init__(self, cache: S3CloudCache):
+        """
+        the passed cache should already have had `cache.load_manifest()`
+        executed. With this satisfied, the attributes
+          - metadata_file_names
+          - file_id_column
+        are populated
+        """
+        expected_metadata = set(["behavior_session_table.csv",
+                                 "ophys_session_table.csv",
+                                 "ophys_experiment_table.csv"])
+        self.cache = cache
+        if cache._manifest.metadata_file_names is None:
+            raise RuntimeError("S3CloudCache object has no metadata "
+                               "file names. BehaviorProjectCloudApi "
+                               "expects a S3CloudCache passed which "
+                               "has already run load_manifest()")
+        cache_metadata = set(cache._manifest.metadata_file_names)
+        if cache_metadata != expected_metadata:
+            raise RuntimeError("expected S3CloudCache object to have "
+                               f"metadata file names: {expected_metadata} "
+                               f"but it has {cache_metadata}")
+        self.logger = logging.getLogger("BehaviorProjectCloudApi")
+        self._get_session_table()
+        self._get_behavior_only_session_table()
+        self._get_experiment_table()
+
+    @staticmethod
+    def from_s3_cache(cache_dir: Union[str, Path],
+                      bucket_name: str) -> "BehaviorProjectCloudApi":
+        cache = S3CloudCache(cache_dir, bucket_name)
+        cache.load_latest_manifest()
+        return BehaviorProjectCloudApi(cache)
+
+    def get_behavior_only_session_data(
+            self, behavior_session_id: int) -> BehaviorSession:
+        """get a BehaviorSession by specifying behavior_session_id
+
+        Parameters
+        ----------
+        behavior_session_id: int
+            the id of the behavior_session
+
+        Returns
+        -------
+        BehaviorSession
+
+        """
+        row = self._behavior_only_session_table.query(
+                f"behavior_session_id=={behavior_session_id}")
+        if row.shape[0] != 1:
+            raise RuntimeError("The behavior_only_session_table should have "
+                               "1 and only 1 entry for a given "
+                               "behavior_session_id. For "
+                               f"{behavior_session_id} "
+                               f" there are {row.shape[0]} entries.")
+        data_path = self.cache.download_data(row.data_file_id.values[0])
```

Comment:
@djkapner there's a case where this can fail due to the design of "behavior_session_table.csv". This table includes both "behavior-only" sessions as well as "behavior ophys sessions". If it is "behavior only" then `row.data_file_id` will be populated. If the session is behavior ophys where we are releasing ophys data, then `row.data_file_id` will be null. The reason is the same as what we discussed for `get_session_data`, because ophys data is at the experiment level.